service/src/test/java/cz/muni/fi/pa165/facade/BottleTypeFacadeTest.java

```diff
@@ -0,0 +1,184 @@
+package cz.muni.fi.pa165.facade;
+
+import cz.muni.fi.pa165.dto.BottleTypeDTO;
+import cz.muni.fi.pa165.entity.BottleType;
+import cz.muni.fi.pa165.entity.Laboratory;
+import cz.muni.fi.pa165.enums.AlcoholType;
+import cz.muni.fi.pa165.service.BottleTypeService;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.List;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertNull;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author mhajas
+ */
+public class BottleTypeFacadeTest extends AbstractFacadeTest{
+
+    @Mock
+    private BottleTypeService bottleTypeService;
+
+    @Autowired
+    @InjectMocks
+    private BottleTypeFacade bottleTypeFacade;
+
+    private BottleType bottleType;
+
+    private BottleTypeDTO bottleTypeDTO;
+
+    private Long bottleTypeId = 1L;
+
+    private String bottleTypeName = "name";
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+
+        bottleTypeFacade = (BottleTypeFacade) unwrapProxy(bottleTypeFacade);
+        ReflectionTestUtils.setField(bottleTypeFacade, "bottleTypeService", bottleTypeService);
+        ReflectionTestUtils.setField(bottleTypeFacade, "beanMappingService", beanMappingService);
+
+        bottleType = new BottleType();
+        bottleType.setName(bottleTypeName);
+
+        bottleTypeDTO = new BottleTypeDTO();
+        bottleTypeDTO.setName(bottleTypeName);
+    }
+
+    @Test
+    public void testFindAll() {
+        when(bottleTypeService.findAll()).thenReturn(Collections.singletonList(bottleType));
+
+        List<BottleTypeDTO> bottleTypes = bottleTypeFacade.findAll();
+
+        assertEquals(bottleType.getName(), bottleTypes.get(0).getName());
```

Comment:
Myslis ze je to dolezite? Akonahle na tom zavolas get(0) hodi to NullPointer ak to bude null. Ci z akeho dovodu myslis?